tests/cnf/ran/ztp/ztp_suite_test.go

```diff
@@ -0,0 +1,56 @@
+package ztp
+
+import (
+	"runtime"
+	"testing"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/openshift-kni/eco-goinfra/pkg/clients"
+	"github.com/openshift-kni/eco-goinfra/pkg/namespace"
+	"github.com/openshift-kni/eco-goinfra/pkg/reportxml"
+	. "github.com/openshift-kni/eco-gotests/tests/cnf/ran/internal/raninittools"
+	"github.com/openshift-kni/eco-gotests/tests/cnf/ran/ztp/internal/helper"
+	"github.com/openshift-kni/eco-gotests/tests/cnf/ran/ztp/internal/tsparams"
+	"github.com/openshift-kni/eco-gotests/tests/internal/reporter"
+)
+
+var _, currentFile, _, _ = runtime.Caller(0)
+
+func TestZtp(t *testing.T) {
+	_, reporterConfig := GinkgoConfiguration()
+	reporterConfig.JUnitReport = RANConfig.GetJunitReportPath(currentFile)
+
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "RAN ZTP Suite", Label(tsparams.Labels...), reporterConfig)
+}
+
+var _ = BeforeSuite(func() {
+	err := helper.GetArgoCdAppGitDetails()
+	Expect(err).ToNot(HaveOccurred(), "Failed to get current data from ArgoCD")
+
+	By("deleting and recreating ZTP test namespace to ensure a blank state")
```

Comment:
I would approach first with Exists() check and then delete ztp test namespace if exists

Reply:
@mpmaruthu Namespace is one of the resources where [deleting returns nil instead of an error if it does not exist](https://github.com/openshift-kni/eco-goinfra/blob/main/pkg/namespace/namespace.go#L158) so it shouldn't cause errors in this case